### ISSUE TITLE
GitHub transition post-tasks

### ISSUE THREAD

**Issue by robertoschwald** [psi-probe/psi-probe#452]
Some issues are still open since the transition to GitHub

Master branch:
Readme.txt should be moved to README.md and updated to reflect the new home

wiki branch:
Is it planned to use https://pages.github.com as the wiki provider? Then you should move this branch to "gh-pages", add a CNAME file and publish. See https://help.github.com/articles/creating-project-pages-manually/ and https://help.github.com/articles/setting-up-a-custom-domain-with-github-pages/

It might be a good idea to state the published wiki url in README.md of master.

Binary distribution
I recommend using bintray to be able to generate and serve binary release war / zip files.

**Comment by MALfunction84** [comment]
Thanks, Robert. All good suggestions.

**Comment by MALfunction84** [comment]
With apologies to forkers, I just rebased and force-pushed to correct problems with the initial import.

**Comment by jander99** [comment]
Might be prudent at least for a bit to add a few people as contributors to this project, rather than have several forks and having to figure out merging later on, for tasks and other clean ups related to the google code migration.  Maybe add a section to the readme for this on how to "interview" for a contrib position.

**Comment by robertoschwald** [comment]
Pull requests?

**Comment by jander99** [comment]
I'm talking about assisting with closing issues, properly tagging them, and making sure all the metadata is cleaned up.  Code stuff can certainly be done with pull requests from forks, however MALfunction then becomes the bottleneck for bugfixes and new releases in the real project, rather than someone else's fork. 

I don't want to blast MALfunction's dedication to the project when it was hosted at Google, but releases and bugfixes were few and far between.  I think if a few more people were able to assist then the project could really take off.  It's just a thought and we can definitely discuss it either in this issue thread or somewhere else.  I'm just hoping the psi-probe community can take some of the burden from one maintainer.

**Comment by MALfunction84** [comment]
@jander99 It's a fair criticism, and the learning curve is working against me. The transition to GitHub was also not timed very well in my own schedule. However, GitHub should speed up merging contributions.

**Comment by robertoschwald** [comment]
It seems the Google Code download page has gone as well, so how do people download releases for now? 
Please attach binary releases of probe to the GH releases. https://github.com/blog/1547-release-your-software

**Comment by MALfunction84** [comment]
The rebase broke all the release branches, and the tags were all tied to release branches. I removed all the tags, and now I want to rebuild to branches. Should I rebase each branch too?  Apologies for my git ignorance.

**Comment by robertoschwald** [comment]
If you add the tags again, the GH releases should be rebuilt. Afterwards you can attach the war zip binaries to the releases.  Forced push is evil ;-)

**Comment by MALfunction84** [comment]
Heh, so I'm learning. I thought it would be fine here at the beginning to clean up after the importer.

So I want to fix the branches before creating all the tags again. Any suggestions?

**Comment by robertoschwald** [comment]
I suggest you checkout the "wiki" branch locally, then create a branch "gh-pages" of it and push this branch to origin. If thats done, you can remove the "wiki" branch. The gh-pages branch is needed for the github project wiki pages, later. See https://help.github.com/articles/creating-project-pages-manually/.

Never perform a forced push. You will hose the git history, therefore everybody forked the repo needs to throw it away. As long as you perform pull, branch, commit, push, you are safe.

**Comment by robertoschwald** [comment]
If you are on OSX, I suggest using "Tower", which helps a lot in working with git repos.

**Comment by MALfunction84** [comment]
Sorry, let's set the wiki aside for a moment. I'm talking about releases right now.

There are several branches related to releases. 1.7, 2.0, 2.1, 2.2, 2.3... These corresponded to minor version number release families, according to the [Release Branch pattern in Subversion](http://svnbook.red-bean.com/en/1.7/svn.branchmerge.commonpatterns.html#svn.branchmerge.commonpatterns.release) (e.g. the 2.0 branch contains all the fixes for any 2.0.x release). Then each tag corresponds to an actual release (e.g. 2.0.0) from its minor version branch.  When I rebased master to fix the import, it changed all the revision fingerprints, so those branches are branching from revisions that no longer exist. *That's* what I'm trying to fix. Then I'll tag each of those branches for their releases.

I'm on Windows.

**Comment by robertoschwald** [comment]
np. For me, your first push looked ok. All the branches and tags were there. See https://github.com/robertoschwald/psi-probe/ (which is a fork after you pushed the first time).

**Comment by MALfunction84** [comment]
I couldn't claim authorship of those revisions, presumably because the author e-mail address is case-sensitive. The issue numbers also didn't link correctly, and that problem persists even now.

Even if I were willing to undo, I'm not sure I can.

**Comment by MALfunction84** [comment]
Moved the wiki pages to gh-pages. Still some work to do there though.

**Comment by MALfunction84** [comment]
I fixed all the branches. Yes, this involved a forced push again, but it was the only way to fix the orphaned branches. It's probably for the best anyway, since the import failed to preserve the actual branch/tag structure of the project.

Speaking of tags, I also fixed the tags, tagging the release family branches with release tags (e.g. 2.3.3 is a tag of a revision on the 2.3 branch, like it was before). Also uploaded the original distributed builds from my own archive to each tag.
